onnxruntime/core/providers/cuda/tensor/transpose_impl.cu

```diff
@@ -79,22 +79,39 @@ Status Transpose3DImpl(cudaStream_t stream, size_t element_size,
   return Status::OK();
 }
 
-template <int element_size>
```

Comment:
binary size reduction: this method needn't really be templated at all

Reply:
the purpose is to optimize the the division below by replacing it with shift instruction during compilation because the division is too expensive.
  CUDA_LONG input_index = (blockIdx.y * input_strides[0] +
                           blockIdx.x * input_strides[1] +
                           threadIdx.y * input_strides[2]) **/
                              (4 * sizeof(int) / element_size)** +
                          threadIdx.x * input_strides[3];

Reply:
Makes sense, fixed.